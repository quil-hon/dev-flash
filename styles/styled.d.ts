import { AppTheme } from "./theme";

declare module "styled-components/native" {
  // DefaultTheme is a empty interface
  // eslint-disable-next-line
  export interface DefaultTheme extends AppTheme {}
}
