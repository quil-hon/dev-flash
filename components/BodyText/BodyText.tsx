import { TextStyle } from "react-native";
import styled from "styled-components/native";

export interface TextBodyProps {
  size?: "normal" | "small" | "x-small";
  variant?: "normal" | "strong" | "highlight";
}

const sizeToFontSize = {
  normal: 16,
  small: 14,
  "x-small": 11,
};

const sizeToLineHeight = {
  normal: 20,
  small: 16,
  "x-small": 14,
};

export const TextBody = styled.Text<TextBodyProps>`
  color: ${(props) => props.theme.colors.dark[200]};
  font-weight: ${(props) =>
    props.variant === "strong"
      ? "700"
      : (props.style as TextStyle)?.fontWeight ?? "normal"};
  font-style: ${(props) =>
    props.variant === "highlight"
      ? "italic"
      : (props.style as TextStyle)?.fontStyle ?? "normal"};
  font-size: ${(props) => sizeToFontSize[props.size ?? "normal"]};
  line-height: ${(props) => sizeToLineHeight[props.size ?? "normal"]};
`;

export default TextBody;
