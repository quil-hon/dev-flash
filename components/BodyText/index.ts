export { default } from "./BodyText";
export { TextBodyProps } from "./BodyText";
