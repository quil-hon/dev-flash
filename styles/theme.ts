import { range } from "../utils/range";

const base = 4;
const sizes = [];
for (const size of range(96)) {
  sizes.push(size * base);
}

export const lightTheme = {
  shadows: {
    medium: [
      {
        shadowColor: "#000",
        shadowOffset: { x: 0, y: 3 },
        shadowRadius: 6,
        shadowOpacity: 8,
      },
    ],
  },
  typography: {
    heading: {
      1: {
        fontSize: 40,
        fontWeight: 700,
        lineHeight: 48,
      },
      2: {
        fontSize: 33,
        fontWeight: 700,
        lineHeight: 40,
      },
      3: {
        fontSize: 27,
        fontWeight: "normal",
        lineHeight: 32,
      },
      4: {
        fontSize: 20,
        fontWeight: 700,
        lineHeight: 24,
      },
      5: {
        fontSize: 20,
        fontWeight: "normal",
        lineHeight: 24,
      },
      6: {
        fontSize: 16,
        fontWeight: "normal",
        lineHeight: 19,
      },
    },
    body: {
      fontFamily: "",
    },
  },
  colors: {
    background: "#F5F5F5",
    primary: {
      50: "#6D8C83",
      100: "#0B3F30",
      200: "#07261D",
    },
    secondary: {
      50: "#F1CB93",
      100: "#E39828",
      200: "#A5752D",
    },
    dark: {
      50: "#AEAEB0",
      100: "#6B6B76",
      200: "#18181B",
    },
    white: {
      50: "#FFF",
      100: "#F1F1F5",
      200: "#EAEAF1",
    },
    error: {
      50: "#F29A7F",
      100: "#EA572A",
      200: "#AC401F",
    },
  },
  space: sizes,
};

export type AppTheme = typeof lightTheme;
