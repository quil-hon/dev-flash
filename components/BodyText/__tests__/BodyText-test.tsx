import React from "react";
import renderer from "react-test-renderer";
import { ThemeProvider } from "styled-components/native";
import { lightTheme } from "../../../styles/theme";
import BodyText from "../BodyText";

test("renders normal BodyText", () => {
  const component = (
    <ThemeProvider theme={lightTheme}>
      <BodyText>This is normal body text</BodyText>
    </ThemeProvider>
  );
  const tree = renderer.create(component).toJSON();
  expect(tree).toMatchSnapshot();
});

test("renders small size BodyText", () => {
  const component = (
    <ThemeProvider theme={lightTheme}>
      <BodyText size="small">This is small body text</BodyText>
    </ThemeProvider>
  );
  const tree = renderer.create(component).toJSON();
  expect(tree).toMatchSnapshot();
});

test("renders x-small size BodyText", () => {
  const component = (
    <ThemeProvider theme={lightTheme}>
      <BodyText size="x-small">This is x-small body text</BodyText>
    </ThemeProvider>
  );
  const tree = renderer.create(component).toJSON();
  expect(tree).toMatchSnapshot();
});
