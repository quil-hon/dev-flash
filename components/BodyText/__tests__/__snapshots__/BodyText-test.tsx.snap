// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders normal BodyText 1`] = `
<Text
  style={
    Array [
      Object {
        "color": "#18181B",
        "fontSize": 16,
        "fontStyle": "normal",
        "fontWeight": "normal",
        "lineHeight": 20,
      },
    ]
  }
>
  This is normal body text
</Text>
`;

exports[`renders small size BodyText 1`] = `
<Text
  size="small"
  style={
    Array [
      Object {
        "color": "#18181B",
        "fontSize": 14,
        "fontStyle": "normal",
        "fontWeight": "normal",
        "lineHeight": 16,
      },
    ]
  }
>
  This is small body text
</Text>
`;

exports[`renders x-small size BodyText 1`] = `
<Text
  size="x-small"
  style={
    Array [
      Object {
        "color": "#18181B",
        "fontSize": 11,
        "fontStyle": "normal",
        "fontWeight": "normal",
        "lineHeight": 14,
      },
    ]
  }
>
  This is x-small body text
</Text>
`;
